north_api/src/api.rs

```diff
@@ -0,0 +1,116 @@
+use serde::{Deserialize, Serialize};
+
+type ContainerId = String;
+
+#[derive(Clone, Eq, PartialEq, Debug, Serialize, Deserialize)]
+pub enum Request {
+    ListContainers,
+    RunningContainers,
+    StartContainer(ContainerId),
+    StopContainer(Option<ContainerId>),
+    UninstallContainer(ContainerId),
+    Update(String),
+    Shutdown,
+    GetSettings,
+    GetVersions,
+}
+
+#[derive(Clone, Eq, PartialEq, Debug, Serialize, Deserialize)]
+pub struct RequestMsg {
+    pub request: Request,
+    pub request_id: u8, // used to match response with a request
+}
+
+#[derive(Clone, Eq, PartialEq, Debug, Serialize, Deserialize)]
+pub enum ContainerType {
+    APP,
+    RESOURCE,
+    SYSTEM,
+}
+
+#[derive(Clone, Eq, PartialEq, Debug, Serialize, Deserialize)]
+pub struct ContainerInfo {
+    pub name: String,
+    pub version: String,
+    pub running_pid: Option<u32>,
+    pub container_type: ContainerType,
+}
+
+#[derive(Clone, Eq, PartialEq, Debug, Serialize, Deserialize)]
+pub struct MemoryInfo {
+    pub size: u64,
+    pub resident: u64,
+    pub shared: u64,
+    pub text: u64,
+    pub data: u64,
+}
+
+#[derive(Clone, Eq, PartialEq, Debug, Serialize, Deserialize)]
+pub struct ProcessInfo {
+    pub name: String,
+    pub version: String,
+    pub pid: u32,
+    pub mem_info: Option<MemoryInfo>,
+    pub uptime_in_us: Option<u64>,
+}
+
+#[derive(Clone, Eq, PartialEq, Debug, Serialize, Deserialize)]
+pub struct StartupInfo {
+    pub id: String,
+    pub success: Option<String>,
+    pub duration_in_ns: Option<u64>,
+}
+
+#[derive(Clone, Eq, PartialEq, Debug, Serialize, Deserialize)]
+pub struct StopInfo {
+    pub id: String,
+    pub success: Option<String>,
+    pub duration_in_ns: Option<u64>,
+}
+
+#[derive(Clone, Eq, PartialEq, Debug, Serialize, Deserialize)]
+pub struct StatusResponse {
+    pub status_info: Option<String>,
+    pub command_response: Option<String>,
+}
+
+#[derive(Clone, Eq, PartialEq, Debug, Serialize, Deserialize)]
+pub struct UninstallInfo {
+    pub name: String,
+    pub uninstalled_version: Option<String>,
+    pub result: Option<String>,
+}
+
+#[derive(Clone, Eq, PartialEq, Debug, Serialize, Deserialize)]
+pub struct UpdateInfo {
+    pub name: Option<String>,
+    pub old_version: Option<String>,
+    pub new_version: Option<String>,
+}
+
+#[derive(Clone, Eq, PartialEq, Debug, Serialize, Deserialize)]
+pub struct VersionInfo {
+    pub name: Option<String>,
+    pub version: Option<String>,
+    pub architecture: Option<String>,
+}
+
+#[derive(Clone, Eq, PartialEq, Debug, Serialize, Deserialize)]
+pub enum Response {
+    ContainerList(Vec<ContainerInfo>),
```

Comment:
The enum item in `Response` could match exactly the corresponding request ones. This makes it easier to understand.